### PR TITLE
CIF-876 - productSKU and cateogoryID replaced with URL Slug in config…

### PR DESCRIPTION
**Pull request by deepkamal**
…s and path formation

## Description

Bug fix  - productSKU and categoryID replaced by URL Slug in edit dialog and URL formation.

## Related Issue

CIF-876

## Motivation and Context

Bug-fix

## How Has This Been Tested?

Manual testing and Junit test case

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

**Comment by deepkamal** [comment]
> Sorry for the late review. The requested fix was for product slugs and _not_ for categories, because category pages URLs use the category id and not the slug (because Magento's GraphQL API doesn't support fetching a category by slug). I'm a bit worried by that change, how did you test that?

Tested it manually and through test case, I see that in absence of URL Slug for category, the categorypicker uses category ID, and the page is linked to right category in the end, how do we want to proceed about it now?

**Comment by cjelger** [comment]
Please just revert the changes you did for categories, using the ID is correct for categories.

**Comment by deepkamal** [comment]
> Please just revert the changes you did for categories, using the ID is correct for categories.

done, please review further.

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/teaser/CommerceTeaser.java [hunk 1]
```diff
@@ -23,16 +23,14 @@
...
+     * Name of the resource property that stores the Product Slug which is used to build Call-to-Action link for Product Page
      *
-     * @since com.adobe.cq.wcm.core.components.models 12.4.0
      */
-    String PN_ACTION_PRODUCT_SKU = "productSKU";
+    String PN_ACTION_PRODUCT_SLUG = "productSlug";
 
     /**
-     * Name of the resource property that stores Category ID to which is used to build Call-to-Action link for Category Page
+     * Name of the resource property that stores Category Slug which is used to build Call-to-Action link for Category Page
```
Comment (cjelger) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/teaser/CommerceTeaser.java]: please revert

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/_cq_dialog/.content.xml [hunk 2]
```diff
@@ -80,19 +80,21 @@
...
+                                                            selectionId="slug">
                                                             <granite:data
                                                                 jcr:primaryType="nt:unstructured"
                                                                 cmp-teaser-v1-dialog-edit-hook="actionLink"/>
                                                         </link>
                                                         <catLink
+                                                            granite:class="cmp-teaser__editor-actionField"
                                                             jcr:primaryType="nt:unstructured"
                                                             sling:resourceType="commerce/gui/components/common/cifcategoryfield"
                                                             emptyText="Catagory"
```
Comment (cjelger) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/_cq_dialog/.content.xml]: please fix "Catagory"

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/teaser/CommerceTeaser.java (hunk 1 above)
Comment (deepkamal) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/teaser/CommerceTeaser.java]: aah,, you mean fix the typo `category ID to` -> `category ID` , i.e. removing the `to` , or README.md to refer as `...stores Category Slug...` ?

Comment (cjelger) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/teaser/CommerceTeaser.java]: I meant "revert your change renaming ID to Slug", but yes you can also remove the unneeded "to"

to have "Name of the resource property that stores Category ID which"

Comment (herzog31) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/teaser/CommerceTeaser.java]: > Name of the resource property that stores Category ID which is used to build Call-to-Action link for Category Page

Comment (deepkamal) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/teaser/CommerceTeaser.java]: got it, done

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/_cq_dialog/.content.xml (hunk 2 above)
Comment (deepkamal) [reply ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/_cq_dialog/.content.xml]: done

**Comment by cjelger** [comment]
The URL links work, but the styling of the dialog could be improved and we have to add a proxy component in the archetype. I'm merging now, will open other issues.

**Pull request merged**